src/plugins/intel_gna/tests/unit/transformations/gna_pwl.cpp

```diff
@@ -4,109 +4,285 @@
+}
+
+template <typename T>
+inline void GnaPWlTestsFixture<T>::run(bool debug) {
     {
         ngraph::pass::Manager m;
         m.register_pass<ngraph::pass::InitNodeInfo>();
-        m.register_pass<ov::intel_gna::pass::PWLApproximation>(max_error_percent);
-        m.run_passes(func);
-        ASSERT_NO_THROW(check_rt_info(func));
+        m.register_pass<ov::intel_gna::pass::PWLApproximation>(_max_error_percent);
+        m.run_passes(_function_under_test);
+        ASSERT_NO_THROW(check_rt_info(_function_under_test));
     }
 
-    auto shape = func->input().get_node_shared_ptr()->get_output_shape(0);
+    auto shape = _function_under_test->input().get_node_shared_ptr()->get_output_shape(0);
     ov::runtime::TensorVector result(1);
-    std::vector<float> data = CommonTestUtils::generate_float_numbers(ov::shape_size(shape), lower_bound, upper_bound);
+    std::vector<float> data =
+        CommonTestUtils::generate_float_numbers(ov::shape_size(shape), _lower_bound, _upper_bound);
     ov::runtime::Tensor input{ov::element::f32, shape, data.data()};
-    ASSERT_TRUE(func->evaluate(result, ov::runtime::TensorVector{input}));
-
-    ov::runtime::TensorVector result_ref(1);
-    ASSERT_TRUE(reference_func->evaluate(result_ref, ov::runtime::TensorVector{input}));
+    ASSERT_TRUE(_function_under_test->evaluate(result, ov::runtime::TensorVector{input}));
 
     const float* result_data = result[0].data<float>();
-    const float* result_ref_data = result_ref[0].data<float>();
-    for (size_t i = 0; i < result[0].get_size(); i++) {
-        double delta = std::abs(result_data[i] - result_ref_data[i]);
-        ASSERT_TRUE(delta <= max_error_percent);
+    std::vector<float> results(result_data, result_data + result[0].get_size());
+
+    validate_results(data, results, _max_error_percent, debug);
+}
+
+template <typename T>
+inline void GnaPWlTestsFixture<T>::validate_results(const std::vector<float>& input_data,
+                                                    const std::vector<float>& results,
+                                                    double max_error_percent,
+                                                    bool debug_enabled) {
+    ASSERT_FALSE(results.empty());
+
+    std::vector<float> reference_values;
+    std::for_each(input_data.begin(), input_data.end(), [&reference_values, this](const double& x) {
+        reference_values.push_back(_reference_function(x));
+    });
+
+    ASSERT_EQ(results.size(), reference_values.size());
+
+    auto abs_peak_to_peak = count_abs_peak_to_peak();
+    if (debug_enabled) {
+        std::cout << "AbsPeakToPeak: " << abs_peak_to_peak << ", max_error_percent=" << max_error_percent << std::endl;
+    }
+
+    for (int i = 0; i < results.size(); ++i) {
+        double delta = std::abs(static_cast<double>(results[i]) - static_cast<double>(reference_values[i]));
+        double error_percentage = delta / abs_peak_to_peak * 100.0;
+        if (debug_enabled) {
```

Comment:
Can we cleanup debug related stuff from tests?

Reply:
done